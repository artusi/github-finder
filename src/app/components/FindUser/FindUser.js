import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import PropTypes from "prop-types";
import { requestUserUpdate } from "store/state/github";
import { SearchField } from "ui";
import styles from "./FindUser.module.css";

export class FindUser extends Component {
  constructor() {
    super();
    this.requestUser = this.requestUser.bind(this);
  }

  componentDidMount() {
    // Initial request when we don't have any user yet
    this.requestUser(this.props.initialUser);
  }

  requestUser(username) {
    this.props.requestUserUpdate(username);
    this.props.onUpdateUser(username);
  }

  render() {
    return (
      <div className={styles.findUser}>
        <SearchField
          id="findUser"
          defaultValue={this.props.initialUser}
          onClick={this.requestUser}
          prefix="https://github.com/"
          label="Search a GitHub user:"
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  github: state.github
});

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      requestUserUpdate
    },
    dispatch
  );
}

FindUser.propTypes = {
  requestUserUpdate: PropTypes.func.isRequired,
  onUpdateUser: PropTypes.func.isRequired,
  initialUser: PropTypes.string
};
FindUser.defaultProps = {
  initialUser: ""
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FindUser);
