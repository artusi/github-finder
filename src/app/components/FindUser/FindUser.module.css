.findUser {
  padding: 16px;
  border-radius: 12px;
  background: #f2cd5d;
  max-width: 420px;
}

@media (max-width: 400px) {
  .findUser {
    width: 270px;
    margin: 10px auto;
  }
}
