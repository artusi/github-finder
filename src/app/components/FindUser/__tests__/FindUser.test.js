import React from "react";
import { FindUser } from "../FindUser";

describe("<FindUser />", () => {
  it("renders", () => {
    const wrapper = shallow(
      <FindUser requestUserUpdate={() => {}} onUpdateUser={() => {}} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders > with initial user", () => {
    const wrapper = shallow(
      <FindUser
        requestUserUpdate={() => {}}
        onUpdateUser={() => {}}
        initialUser="kakaroto"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
