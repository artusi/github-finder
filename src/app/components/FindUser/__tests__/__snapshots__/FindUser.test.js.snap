// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FindUser /> renders > with initial user 1`] = `
<div
  className="findUser"
>
  <SearchField
    defaultValue="kakaroto"
    id="findUser"
    label="Search a GitHub user:"
    onClick={[Function]}
    prefix="https://github.com/"
  />
</div>
`;

exports[`<FindUser /> renders 1`] = `
<div
  className="findUser"
>
  <SearchField
    defaultValue=""
    id="findUser"
    label="Search a GitHub user:"
    onClick={[Function]}
    prefix="https://github.com/"
  />
</div>
`;
